app/controllers/admin/base_controller.rb

```diff
@@ -1,3 +1,8 @@
 class Admin::BaseController < ApplicationController
+  before_action :check_role
   layout "layouts/application_admin"
+
+  def check_role
+    redirect_to(root_path) unless current_user&.admin?
```

Comment:
thêm message để thân thiện ng dùng bé bì, chẳng hạn, bạn ko có quyền truy câp thông tin này

Reply:
thay vì dùng toán tử an toàn `&.` em nên bổ sung việc check user đã được login nhé
nếu chưa login thì di chuyển màn login và kèm message như bạn cần login trươcs khi di chuyển vào màn hình này .... 
meesssage chuẩn và cụ thể để ng dung happy hơn

Reply:
dạ vâng ạ